kernel/src/main/java/org/cristalise/kernel/security/SecurityManager.java

```diff
@@ -124,16 +125,21 @@ public Subject getSubject(String principal) {
     }
 
     /**
-     * 
+     * Loads shiro.ini file from a file or from the classpath (default)
+     * TODO: replace the use of IniSecurityManagerFactory with shiro Environment initialization
      */
     public void setupShiro() {
-        //TODO: replace this with shiro Environment initialization
-        Factory<org.apache.shiro.mgt.SecurityManager> factory = new IniSecurityManagerFactory("classpath:shiro.ini");
+        String shiroIni = Gateway.getProperties().getString("Shiro.iniFile");
+
+        if (StringUtils.isBlank(shiroIni)) shiroIni = "classpath:shiro.ini";
+        else                               shiroIni = "file:" + shiroIni;

```

Comment:
Use CLASSPATH_PREFIX and FILE_PREFIX constants from https://shiro.apache.org/static/1.3.2/apidocs/org/apache/shiro/io/ResourceUtils.html#CLASSPATH_PREFIX

Reply:
I think this will hinder code readability in  the sense, that it will not be easy to know the actual string that was constructed. Also using string constant is recommended when the string is used many times. but this is not the case.